test/functional/change_stream_tests.js

```diff
@@ -1677,4 +1677,133 @@ describe('Change Streams', function() {
         .then(() => finish(), err => finish(err));
     }
   });
+
+  describe('should properly handle a changeStream event being processed mid-close', function() {
+    it('when invoked with promises', {
+      metadata: { requires: { topology: 'replicaset', mongodb: '>=3.5.10' } },
+      test: function() {
+        function read(coll) {
+          const changeStream = coll.watch();
+          return Promise.resolve()
+            .then(() => changeStream.next())
+            .then(() => changeStream.next())
+            .then(() => {
+              const closeP = Promise.resolve().then(() => changeStream.close());
```

Comment:
Interesting, does `changeStream.close()` not return a `Promise`? Looks like [it should](https://github.com/mongodb/node-mongodb-native/blob/master/lib/change_stream.js#L153) to me. This should at least be: `Promise.resolve(changeStream.close())`

Reply:
This is actually a relic of when I was trying to enqueue `changeStream.close` as a microTask. I can remove it.